File: app/src/main/java/com/blog/demo/component/service/AidlService.java
package com.blog.demo.component.service;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import androidx.annotation.Nullable;

public class AidlService extends Service {
    private AidlInterface mInterface = new AidlInterface();

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return mInterface;
    }

    private static class AidlInterface extends IAidlInterface.Stub {

        @Override
        public int add(int val1, int val2) {
            return val1 * 10 + val2;
        }

    }
}

File: app/src/main/java/com/blog/demo/component/service/AidlService.kt
package com.blog.demo.component.service

import android.app.Service
import android.content.Intent
import android.os.IBinder
import com.blog.demo.LogTool

class AidlService : Service() {
    private var mInterface = AidlInterface()

    override fun onBind(intent: Intent?): IBinder? {
        return mInterface
    }

    private inner class AidlInterface : IAidlInterface.Stub() {

        override fun add(val1: Int, val2: Int): Int {
            LogTool.logi("AidlService", toString())
            return val1 * 10 + val2
        }

    }

}